config/kibana.yml

```diff
@@ -1,8 +1,11 @@
 # Kibana is served by a back end server. This setting specifies the port to use.
 # server.port: 5601
 
-# This setting specifies the IP address of the back end server.
-# server.host: "0.0.0.0"
+# Specifies the host name to which the Kibana server will bind.
+# The default is 'localhost', which usually means remote machines will not be able to connect.
+# To allow connections from remote users, set this parameter to IP or the DNS-name of the machine that is running Kibana.
+# For example: if remote users need to connect to Kibana via http://foobar:5601, set this parameter to 'foobar'.
+# server.host: "localhost"
```

Comment:
I wouldn't say so. It's the destination of the public API endpoint, the name the end-user will use to connect. The actual resolution of that is not all that relevant. I can setup resolution for 'foobar' in my host-file and point it to the loopback interface. Or `foobar` resolution could happen on the network. Either way, in both cases Kibana needs to just bind to `foobar`. Similarly for an IP, for Kibana it does not matter which interface is mapped to which address, as long as the name matches the incoming request. 

We probably want to discourage mapping to IPs too. If we map Kibana to the IP assigned to the public network interface, but people connect with a DNS-name that resolves to that host, the app-server (`hapi `in this case) will ignore it.

Perhaps there is intricacies in a VM setup that I am not aware of though.

We should simplify the doc though, to reduce ambiguity. Something like (?)
`
Specifies the public hostname or IP address of the server.
The default is `localhost` which means remote machines will not be able to connect.
To allow connections from remote users, set this parameter to the public host name.
...
`